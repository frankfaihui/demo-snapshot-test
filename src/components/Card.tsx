import React, { ReactNode } from "react";
import styled from "@emotion/styled";
import BaseButton from "./BaseButton";

// Card props
export interface CardProps {
  title: string;
  description?: string;
  buttonText?: string;
  onButtonClick?: () => void;
  children?: ReactNode;
  className?: string;
}

const StyledCard = styled.div({
  backgroundColor: "white",
  borderRadius: "0.5rem",
  boxShadow: "0 1px 3px 0 rgba(0, 0, 0, 0.1), 0 1px 2px 0 rgba(0, 0, 0, 0.06)",
  padding: "1.5rem",
  display: "flex",
  flexDirection: "column",
  gap: "1rem",
});

const CardTitle = styled.h3({
  fontSize: "1.25rem",
  fontWeight: 600,
  color: "#1a202c",
  margin: 0,
});

const CardDescription = styled.p({
  fontSize: "1rem",
  color: "#4a5568",
  margin: 0,
});

const CardFooter = styled.div({
  marginTop: "auto",
  display: "flex",
  justifyContent: "flex-end",
});

export default function Card({
  title,
  description,
  buttonText,
  onButtonClick,
  children,
  className,
}: CardProps) {
  return (
    <StyledCard className={className}>
      <CardTitle>{title}</CardTitle>
      {description && <CardDescription>{description}</CardDescription>}
      {children}
      {buttonText && (
        <CardFooter>
          <BaseButton onClick={onButtonClick}>{buttonText}</BaseButton>
        </CardFooter>
      )}
    </StyledCard>
  );
} 