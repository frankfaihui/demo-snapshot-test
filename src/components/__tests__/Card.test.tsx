import React from 'react';
import { render } from '@testing-library/react';
import Card from '../Card';

describe('Card Component Snapshot', () => {
  it('renders correctly with all props', () => {
    const { container } = render(
      <Card 
        title="Test Card Title" 
        description="This is a test description for the card component"
        buttonText="Click Me"
        onButtonClick={() => {}}
      >
        <div data-testid="test-child">Card children content</div>
      </Card>
    );
    
    expect(container).toMatchSnapshot();
  });
}); 