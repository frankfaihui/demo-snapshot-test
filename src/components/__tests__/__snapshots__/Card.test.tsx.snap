// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component Snapshot renders correctly with all props 1`] = `
.emotion-0 {
  background-color: white;
  border-radius: 0.5rem;
  box-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.1),0 1px 2px 0 rgba(0, 0, 0, 0.06);
  padding: 1.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1rem;
}

.emotion-1 {
  font-size: 1.25rem;
  font-weight: 600;
  color: #1a202c;
  margin: 0;
}

.emotion-2 {
  font-size: 1rem;
  color: #4a5568;
  margin: 0;
}

.emotion-3 {
  margin-top: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
}

.emotion-4 {
  display: block;
  border: 1px solid #e2e8f0;
  color: black;
  padding: 10px;
  background-color: white;
}

<div>
  <div
    class="emotion-0"
  >
    <h3
      class="emotion-1"
    >
      Test Card Title
    </h3>
    <p
      class="emotion-2"
    >
      This is a test description for the card component
    </p>
    <div
      data-testid="test-child"
    >
      Card children content
    </div>
    <div
      class="emotion-3"
    >
      <button
        class="emotion-4"
      >
        Click Me
      </button>
    </div>
  </div>
</div>
`;
